File: Amazon/ServerSide/Amazon.Solution/Amazon.Core/Entities/Review.cs
using Amazon.Core.Entities.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Amazon.Core.Entities
{
	public class Review : BaseEntity
	{
		[Range(1, 5)]
		public int Rating { get; set; }
		[MaxLength(100)]
		public string ReviewHeadLine { get; set; }
		[MaxLength(1000)]
		public string ReviewText { get; set; }
		public DateTime ReviewDate { get; set; } = DateTime.Now;
		public int ProductId { get; set; }
		public virtual Product Product { get; set; }
		[ForeignKey("AppUser")]
		public string AppUserName { get; set; }
		public string AppUserEmail { get; set; }
	}
}

File: Amazon/ClientSide/Amazon.ClientSide/src/app/Models/review.ts

export class Review {
    id: number;
    rating: number;
    reviewHeadLine: string;
    reviewText: string;
    reviewDate: Date;
    productId: number;
    appUserName: string;
    appUserEmail: string;

    constructor(
        id: number,
        rating: number,
        reviewHeadLine: string,
        reviewText: string,
        reviewDate: Date,
        productId: number,
        appUserName: string,
        appUserEmail: string
    ) {
        this.id = id
        this.rating = rating
        this.reviewHeadLine = reviewHeadLine
        this.reviewText = reviewText
        this.reviewDate = reviewDate
        this.productId = productId
        this.appUserName = appUserName
        this.appUserEmail = appUserEmail
    }
}
